docs/CHAIN_CONFIG.md

```diff
@@ -0,0 +1,192 @@
+            l2GasCalculator: chainParams.l2GasCalculator,
+            // ... other parameters
+        });
+    }
+}
+```
+
+### Helper Functions
+
+```solidity
+// Get chain name
+string memory chainName = ChainConfig.getChainName(block.chainid);
+
+// Check if L2 gas calculator is required
+bool requiresCalc = ChainConfig.requiresL2GasCalculator(block.chainid);
+
+// Get surcharge rates
+ChainConfig.ChainParameters memory params = ChainConfig.getChainParameters(block.chainid);
+uint256 atlasSurcharge = params.atlasSurchargeRate;    // in basis points (1000 = 10%)
+uint256 bundlerSurcharge = params.bundlerSurchargeRate; // in basis points
+```
+
+## Adding New Chains
+
+To add support for a new chain:
+
+1. Add the chain configuration in `ChainConfig.sol`:
+```solidity
+} else if (chainId == YOUR_CHAIN_ID) {
+    return ChainParameters({
+        escrowDuration: BLOCKS,  // Calculate based on block time
+        l2GasCalculator: address(0), // Or specific address if needed
+        atlasSurchargeRate: 1000,  // 10% default
+        bundlerSurchargeRate: 1000, // 10% default
+        name: "YOUR_CHAIN_NAME"
+    });
+}
+```
+
+2. Update `requiresL2GasCalculator()` if the chain needs a gas calculator:
+```solidity
+function requiresL2GasCalculator(uint256 chainId) internal pure returns (bool) {
+    return chainId == 42161 || chainId == 421614 || // Arbitrum chains
+           chainId == YOUR_CHAIN_ID; // Add your chain if needed
+}
+```
+
+## Escrow Duration Calculation
+
+Escrow duration is calculated to achieve approximately 30-35 seconds of real time:
+
+- **Ethereum/Sepolia**: 64 blocks × 12s = ~768s (adjusted for network conditions)
+- **Polygon/Amoy**: 64 blocks × 2s = ~128s (adjusted for faster finality)
+- **OP Stack (Optimism/Base)**: 16 blocks × 2s = ~32s
+- **Arbitrum**: 128 blocks × 250ms = ~32s
+- **Hyperliquid/Unichain**: 30 blocks × 1s = ~30s
+- **Berachain**: 15 blocks × 5s = ~75s (Cosmos-based consensus)
+
+## L2 Gas Calculator
+
+L2 Gas Calculator requirements by chain type:
+- **OP Stack chains (Base, Optimism, Unichain)**: Require L2 gas calculator for proper gas accounting
+  - **Automatic Deployment**: The Atlas deployment script automatically deploys an L2 gas calculator if one is required but not configured
+  - **Manual Deployment** (optional): Can be deployed separately using `script/deploy-gas-calculator-op-stack.s.sol`
+- **Arbitrum**: Not supported on main branch - use branches `arbitrum/atlas-v1.6.1` or `arbitrum/atlas-v1.7-exp`
+- **Other chains (Ethereum, Polygon, Hyperliquid, Berachain)**: Standard gas calculation, no L2 calculator needed
+
+## Arbitrum Deployment
+
+Arbitrum requires special handling due to its unique architecture and is maintained on separate branches:
+
+### Available Arbitrum Branches
+- `arbitrum/atlas-v1.6.1` - Stable version for Arbitrum One and Arbitrum Sepolia
+- `arbitrum/atlas-v1.7-exp` - Experimental version with latest features
```

Comment:
I also don't think v1.7 experimental is ready for any kind of mainnet deployment, so we probably shouldn't advise using it